setup_dependencies.sh

```diff
@@ -66,227 +58,139 @@ function detect_distribution ()
+    local cmake_sha256="020812a9f87293482cec51fdf44f41cc47e794de568f945a8175549d997e1760"
+    local cmake_executable="${cmake_dir}/bin/cmake"
 
-    if [ -f ${OPT_DIR}/cmake/bin/cmake ]
+    if [ ! -x "${cmake_executable}" ]
     then
-        local cmake_version=$(${OPT_DIR}/cmake/bin/cmake --version | awk /1/{'print $3'})
-        info "Found cmake  in ${OPT_DIR}, version: ${cmake_version}"
-        return 0
-    fi
+        info "Setup cmake package"
 
-    if command -v cmake >/dev/null
-    then
-        # Check existing cmake version
-        local cmake_version=$(cmake --version | awk /1/{'print $3'})
-        local cmake_version_major=$(printf "${cmake_version}" | awk -F '.' {'print $1'})
-        local cmake_version_minor=$(printf "${cmake_version}" | awk -F '.' {'print $2'})
+        if [ -f "${cmake_tar_path}" ]
+        then
+             echo "${cmake_sha256} ${cmake_tar_path}" | sha256sum -c &>/dev/null
+             if [ $? -ne 0 ]
+             then
+                info CHECK sum is wrong: ${cmake_tar_path}
+                info Remove package and re-download it
+                rm "${cmake_tar_path}"
+             fi
+        fi
 
-        if [ "$cmake_version_major" -lt "$MIN_CMAKE_VER_MAJOR" ]
+        if [ ! -f "${cmake_tar_path}" ]
         then
-            info "Insufficent cmake version found"
-            return 1
+            info "Download cmake package"
+            wget -O ${cmake_tar_path} "${cmake_link}"
+            check_result $? "Can't download CMake"
         fi
 
-        if [ "$cmake_version_minor" -lt "$MIN_CMAKE_VER_MINOR" ]
+        echo "${cmake_sha256} ${cmake_tar_path}" | sha256sum -c
+        check_result $? "Invalid SHA256 sum for downloaded file! Aborting."
+
+        info "Extract cmake package"
+        tar --no-same-owner -xzf ${cmake_tar_path} -C ${cmake_dir}/ --strip-components=1
+        check_result $? "Can't unpack CMake tarball"
+
+        if [ "${SUDO_USER}" != "" ]
         then
-            info "Insufficent cmake version found"
-            return 1
+            info "Change owner of ${cmake_dir} to ${SUDO_USER}"
+            chown -R ${SUDO_USER}:${SUDO_USER} ${cmake_dir}
+            check_result $? "Can't change owner of ${cmake_dir} to ${SUDO_USER}"
         fi
 
-        info "Found cmake, version: ${cmake_version}"
-        return 0
+        info "cmake installed to ${cmake_dir}"
+    else
+        info "cmake already installed to ${cmake_dir}"
     fi
+}
+
+function setup_dependencies () {
+    setup_cmake
+}
 
-    info "No cmake found"
-    return 1;
+function get_distribution_pkg_installer () {
+    distro=$(detect_distribution)
+    case "${distro}" in
+    "RHEL7"|"CENTOS7")
+        echo "yum -y install"
+        ;;
+    "FEDORA")
+        echo "dnf -y install"
+        ;;
+    "UBUNTU")
+        echo "apt-get -y install"
+        ;;
+    *)
+        error "Unknown Linux distribution"
```

Comment:
I'd also print here the packages that need to be installed for the user to install it manually.

Reply:
Altough I noticed they are in logs anyway probably, if so, this comment can be ignored